components/org.wso2.carbon.secvault.securevault/src/main/java/org/wso2/carbon/secvault/securevault/internal/SecureVaultComponent.java

```diff
@@ -72,21 +73,35 @@ public void deactivate(BundleContext bundleContext) {
             unbind = "unRegisterSecretRepository"
     )
     protected void registerSecretRepository(SecretRepository secretRepository) {
-        if (secretRepository.getClass().getName().equals(
-                SecureVaultInitializer.getInstance().getSecretRepositoryType())) {
-            logger.debug("Registering secret repository : {}", SecureVaultInitializer.getInstance()
-                    .getSecretRepositoryType());
-            SecureVaultDataHolder.getInstance().setSecretRepository(secretRepository);
-            initializeSecureVault();
+        Path secureVaultYAMLPath = Utils.getCarbonConfigHome()
+                .resolve(SecureVaultConstants.SECURE_VAULT_CONFIG_YAML_FILE_NAME);
+        try {
+            SecureVaultConfiguration secureVaultConfiguration = SecureVaultConfigurationProvider.getInstance()
+                    .getConfiguration().orElseThrow(() -> new SecureVaultException(SECURE_VAULT_CONFIG_ERROR));
+            String secretRepositoryType = secureVaultConfiguration.getSecretRepositoryConfig().getType()
+                    .orElseThrow(() -> new SecureVaultException("Secret repository type is not set"));
```

Comment:
It might be better to mention in the method signature that SecureVaultException could be thrown in this method.